pkg/password/manager.go

```diff
@@ -0,0 +1,33 @@
+package password
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+
+	"github.com/giantswarm/microerror"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type Manager interface {
+	GeneratePassword(length int) (string, error)
+	Hash(plaintext string) (string, error)
+}
+
+type SimpleManager struct {
+}
+
+func (m SimpleManager) GeneratePassword(length int) (string, error) {
+	bytes := make([]byte, length)
+	if _, err := rand.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
+
+func (m SimpleManager) Hash(plaintext string) (string, error) {
+	bytes, err := bcrypt.GenerateFromPassword([]byte(plaintext), 14)
```

Comment:
Why are we hashing the password through bcrypt here ?

Reply:
1: Because it needs to be hashed
2: because nginx does not support a lot of hashes
3: because bcrypt is one of the best security hashes out there

Reply:
I though basic auth where just base64 encoded, maybe I am wrong.

Reply:
Sadly no: https://docs.giantswarm.io/advanced/ingress/configuration/#authentication It requires a hash of the password. In the simplest example, it uses apache `htpasswd` but there is no go implementation to cypher the password so I picked the best in go that nginx supports

Reply:
Ok fine